test/gtest/lattices.cpp

```diff
@@ -546,3 +547,79 @@ TEST(ValTypeLattice, Meet) {
                   Type(HeapType::array, Nullable),
                   Type(HeapType::eq, Nullable));
 }
+
+TEST(SharedLattice, GetBottom) {
+  analysis::Shared<analysis::UInt32> shared{analysis::UInt32{}};
+  EXPECT_EQ(*shared.getBottom(), 0u);
+}
+
+TEST(SharedLattice, Compare) {
+  analysis::Shared<analysis::UInt32> shared{analysis::UInt32{}};
+
+  auto zero = shared.getBottom();
+
+  auto one = zero;
+  shared.join(one, 1);
+
+  auto uno = one;
+  shared.join(uno, 1);
+
+  auto two = one;
+  shared.join(two, 2);
+
+  EXPECT_EQ(shared.compare(zero, zero), analysis::EQUAL);
+  EXPECT_EQ(shared.compare(zero, one), analysis::LESS);
+  EXPECT_EQ(shared.compare(zero, uno), analysis::LESS);
+  EXPECT_EQ(shared.compare(zero, two), analysis::LESS);
+
+  EXPECT_EQ(shared.compare(one, zero), analysis::GREATER);
+  EXPECT_EQ(shared.compare(one, one), analysis::EQUAL);
+  EXPECT_EQ(shared.compare(one, uno), analysis::EQUAL);
```

Comment:
To make sure I follow, `uno` had an extra `join` done on it, but it was of the same value as we had before, so the sequence number was not increased, hence it is equal to `one`?

Reply:
That's right 👍